File: fetch-data/Entity/Channel.py
from Entity.Modes import Modes

class Channel:
    
    def __init__(self):
        self.id = ''
        self.title = ''
        self.subscriber_count = 0
        self.video_count = 0
        self.thumbnail_url = ''
        
    def to_tuple(self, mode):   
        if mode == Modes.CREATE:    
            return (self.title, self.subscriber_count, self.thumbnail_url, self.id, self.video_count)     
        if mode == Modes.UPDATE:
            return (self.title, self.subscriber_count, self.thumbnail_url, self.id, self.video_count, self.title) 
        

File: youtube-data-backend/src/main/java/com/example/youtube_data_project/entity/Channel.java
package com.example.youtube_data_project.entity;

import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.Table;

@Entity
@Table(name="channel")
public class Channel {
	@Id
	private String title;
	private int subscriberCount;
	private int videoCount;
	private String thumbnailUrl;
	private String id;
	
	public Channel() {
		
	}

	public String getTitle() {
		return title;
	}

	public void setTitle(String title) {
		this.title = title;
	}

	public int getSubscriberCount() {
		return subscriberCount;
	}

	public void setSubscriberCount(int subscriberCount) {
		this.subscriberCount = subscriberCount;
	}

	public int getVideoCount() {
		return videoCount;
	}

	public void setVideoCount(int videoCount) {
		this.videoCount = videoCount;
	}

	public String getThumbnailUrl() {
		return thumbnailUrl;
	}

	public void setThumbnailUrl(String thumbnailUrl) {
		this.thumbnailUrl = thumbnailUrl;
	}

	public String getId() {
		return id;
	}

	public void setId(String id) {
		this.id = id;
	}
	
	
}
